### ISSUE TITLE
Double Config Read

### ISSUE THREAD

**Issue by jdthomason** [snhu-coders/noob_snhubot#26]
Bot appears to be reading the config file twice.  This is resulting in the following output when starting the bot:

```
UTC: 1541396454.924078 - Reading configuration in config.yml
UTC: 1541396454.92508 - Reading configuration in config.yml
UTC: 1541396455.564339 - Noob SNHUbot connected and running!
```

With some rudimentary debugging, This issue seems to be related to the ordering of module imports versus module execution.  The order that I've been able to deduce so far is:

1. Config file gets read
2. Condition on line 225 does not allow `main()` to execute because `__name__ == noob_snhubot`
3. The `noob_snhubot` module is then imported into its various places across the commands
4. Config file gets read again
5. The `main()` function is now able to execute because `__name__ == __main__`

Or...  something like that.  I did try moving the associated `try` block and surrounding statements below all of the function definitions, which didn't seem to help anything.  Outside of that, I'm certain I'll just break stuff until I've understood the execution order.

**Comment by gsfellis** [comment]
So you're mostly correct.  This is due to the circular import from various `cmds` back to the `noob_snhubot` script.  `__name__ == "__main__"` only when executed as a script, not called as an import by another module.  However, when the script is imported, it will execute all code at the global level, which is why the code is executed twice.  The first time is actually called when the import occurs from one of the commands, and the second is while the script is executing.

The issue becomes all of the variables defined and assigned in the global scope in `noob_snhubot.py`.  They're used all over the place, and it becomes unwieldy to call `global` in all of the functions they're needed.  Might need a redesign of the primary script to address this.

**Comment by gsfellis** [comment]
Being fixed by #35
